server/src/main/java/org/elasticsearch/repositories/RepositoryData.java

```diff
@@ -944,4 +961,48 @@ private static String parseLegacySnapshotUUID(XContentParser parser) throws IOEx
         return uuid;
     }
 
+    /**
+     * A few details of an individual snapshot stored in the top-level index blob, so they are readily accessible without having to load
+     * the corresponding {@link SnapshotInfo} blob for each snapshot.
+     */
+    public static class SnapshotDetails {
+
+        public static SnapshotDetails EMPTY = new SnapshotDetails(null, null);
+
+        @Nullable // TODO when?
```

Comment:
I actually think never at this point. This is only relevant for very old repository data (pre-5.4ish) if I'm not mistaken. We can probably remove this because we don't fully support reading all repository metadata from repositories that old at this point anyway.